app/models/run_file.rb

```diff
@@ -6,16 +6,24 @@ class RunFile < ActiveRecord::Base
 
   def self.for_file(file)
     if file.respond_to?(:read)
-      RunFile.for_text(file.read)
+      RunFile.for_text(file.read, File.binary?(file.tempfile))
     end
   end
 
-  def self.for_text(file_text)
+  def self.for_text(file_text, binary)
     digest = Digest::SHA256.hexdigest(file_text)
-    where(digest: digest).first_or_create(file: file_text)
+    where(digest: digest).first_or_create(file: binary ? RunFile.unpack_binary(file_text) : file_text)
```

Comment:
I'd prefer the binary stuff be in a separate `for_binary` method so that we're not calling this one with a sometimes-unused argument. Then `for_file` can determine which method it needs to call.